### ISSUE TITLE
Wrapping arithmetic flags should be used for HOS code

### ISSUE THREAD

**Issue by v1993** [yuzu-emu/yuzu#6913]
Originally noted on [discord](https://discord.com/channels/398318088170242053/402002111241388032/877946192678486016) by atmosphere dev:
![Screenshot_20210822_194238](https://user-images.githubusercontent.com/16255906/130363168-f36d15b7-6ba0-40d9-bed9-313093814a9d.png)

Would be happy to do this myself, but I have no idea what flags MSVC have for this nor want to do anything related to MSVC in general.

**Comment by lioncash** [comment]
MSVC doesn't have a documented equivalent of -fwrapv. I also don't particularly think we should be sectioning off a part of the code to be using flags that enable non-standard C++ behavior, given we want the emulator to remain maximally portable (which already falls apart, well, when MSVC or any other compiler that doesn't support an equivalent comes along).

I'd much rather prefer us setting up something along the lines of automated fuzzing with asan/ubsan.

**Comment by v1993** [comment]
I agree that it would be the best course of action, but it means that code originating from Atmosphere should be double-checked for things relying on wrapping and modified when it does. Should I submit patch for known issues of such type?

**Comment by lioncash** [comment]
Yeah, absolutely.

**Comment by FernandoS27** [comment]
We should but I don't know any modern CPU that uses anything different to two-complements.

**Comment by v1993** [comment]
> We should but I don't know any modern CPU that uses anything different to two-complements.

Here we go again. Yes, CPUs all do that. As long as compiler transforms code directly into instructions, which it sometimes does not, especially with things like overflow checks. Yes, I've checked with godbolt.

**Comment by FernandoS27** [comment]
hmn, feel free to submit a PR for clang/gcc. Not sure if there's a flag on MSVC.

**Comment by lioncash** [comment]
There is no flag for equivalent behavior. We shouldn't be enabling it for GCC/clang, because then we can end up with scenarios where code behaves differently depending on which compiler it was built with.
